### ISSUE TITLE
2024.12.1 breaks entity update 

### ISSUE THREAD

**Issue by dettofatto** [AlessandroTischer/tuya_climate_mult_edit#5]
With the last update this great extension has some issues with entity update.
I'm alone?

**Comment by AlessandroTischer** [comment]
Hi! What do you mean by that? What's the problem? Mine is still working correctly

**Comment by dettofatto** [comment]
With the new update I've observed that my thermostat doesn't update the temperature, but only when I reload your extension..
Maybe it is my fault, but in any case, I'll wait another update and I restored my backup!

**Comment by amoranvalero** [comment]
It happens to me. I use offical tuya and yours integration for the thermostat with has an uncorrected temperature. Last week your integration worked like a charm anb core Tuya failed to update all devices state. Now this week it is the other way round. Thermostat with your integration does not reflect on/off state or target temperature when modified. Tuya core integration works now. Bit strange but could be both integrations cannot longer live together?

**Comment by amoranvalero** [comment]
> @amoranvalero thanks for the feedback. 
> @dettofatto do you use the official integration too?
> 
> I'm pretty sure it should not matter, since the two should be completely disjoint, but it's worth checking.
> Unfortunately I'm not noticing any misbehaviour, thus it's difficult to understand what's the problem without any log.
> Can you enable debug logging while you see the failures and share the log? 
> ![image](https://github.com/user-attachments/assets/614cafc5-efcf-47cf-97fc-044a048b8230)
> Thank you.

Registrador: homeassistant.helpers.dispatcher
Fuente: util/logging.py:95
Ocurrió por primera vez: 11:16:12 (2 ocurrencias)
Última vez registrado: 11:16:42

Exception in async_write_ha_state when dispatching 'tuya_entry_update_bfcf350ba14c3fab54qsht': (['switch'],) TypeError: Entity.async_write_ha_state() takes 1 positional argument but 2 were given

**Comment by amoranvalero** [comment]
This once I turn on the thermostat which actually turns on. But the state is not logged.

As I said this behaviour happened to tuya official integration whereas your working properly. Yesterday it turned the other way round tuya official integration is correctly logging the change but yours not. That is the reason I guess they cannot both live together.

**Comment by AlessandroTischer** [comment]
> This once I turn on the thermostat which actually turns on. But the state is not logged.
> 
> As I said this behaviour happened to tuya official integration whereas your working properly. Yesterday it turned the other way round tuya official integration is correctly logging the change but yours not. That is the reason I guess they cannot both live together.

It's very strange. I updated the integration with the latest source code. Maybe it's somehow fixed? Could you check it?

**Comment by amoranvalero** [comment]
> > This once I turn on the thermostat which actually turns on. But the state is not logged.
> > 
> > As I said this behaviour happened to tuya official integration whereas your working properly. Yesterday it turned the other way round tuya official integration is correctly logging the change but yours not. That is the reason I guess they cannot both live together.
> 
> It's very strange. I updated the integration with the latest source code. Maybe it's somehow fixed? Could you check it?

😲 WOW now working like a charm an both tuya official and your integration working both. Thanks

**Comment by dettofatto** [comment]
I'll check in the next days! Thank you for your help and work!

**Comment by Mactomik** [comment]
I have same issue. Similar setup TUYA official and this integration installed side by side. I am having following error in logs: 2024-12-17 16:58:49.314 ERROR (MainThread) [homeassistant.util.logging] Exception in _handle_state_update when dispatching 'tuya_entry_update_bfa55087baee4f4e6chbqj': ()
TypeError: TuyaEntity._handle_state_update() missing 1 required positional argument: 'updated_status_properties'

**Comment by AlessandroTischer** [comment]
> I have same issue. Similar setup TUYA official and this integration installed side by side. I am having following error in logs: 2024-12-17 16:58:49.314 ERROR (MainThread) [homeassistant.util.logging] Exception in _handle_state_update when dispatching 'tuya_entry_update_bfa55087baee4f4e6chbqj': () TypeError: TuyaEntity._handle_state_update() missing 1 required positional argument: 'updated_status_properties'

Are you using the latest version?

**Comment by Mactomik** [comment]
> > I have same issue. Similar setup TUYA official and this integration installed side by side. I am having following error in logs: 2024-12-17 16:58:49.314 ERROR (MainThread) [homeassistant.util.logging] Exception in _handle_state_update when dispatching 'tuya_entry_update_bfa55087baee4f4e6chbqj': () TypeError: TuyaEntity._handle_state_update() missing 1 required positional argument: 'updated_status_properties'
> 
> Are you using the latest version?

Yes, I have the latest version tuyaclimatemultiedit. The issue started after installing the 12.2024 update from hacks. I am on hass core 11.4 version, not hass 12.1. However the behavior is very similar. I had no thermostat update few days from update. Today after posting here I reinstalled, reloaded and ended up getting the thermostat updates to work however then official tuya devices (i have some lights) now stoped responding.

**Comment by AlessandroTischer** [comment]
> > > I have same issue. Similar setup TUYA official and this integration installed side by side. I am having following error in logs: 2024-12-17 16:58:49.314 ERROR (MainThread) [homeassistant.util.logging] Exception in _handle_state_update when dispatching 'tuya_entry_update_bfa55087baee4f4e6chbqj': () TypeError: TuyaEntity._handle_state_update() missing 1 required positional argument: 'updated_status_properties'
> > 
> > Are you using the latest version?
> 
> Yes, I have the latest version tuyaclimatemultiedit. The issue started after installing the 12.2024 update from hacks. I am on hass core 11.4 version, not hass 12.1. However the behavior is very similar. I had no thermostat update few days from update. Today after posting here I reinstalled, reloaded and ended up getting the thermostat updates to work however then official tuya devices (i have some lights) now stoped responding.
> 

I suppose they changed something with December update then. I would say, use older version until you update HA.

**Comment by amoranvalero** [comment]
Hello @AlessandroTischer HA last update (2025.3.0) is reproducing the issues with the climate integration. Status of thermostat and climate are not updated and information of the controls are messed up. Can you check or update your dev to last core update? The issue is similar to the previous identified in present thread. Thanks a lot

**Comment by AlessandroTischer** [comment]
Hello everyone. I need some days (maybe a week) before having the time to give this a proper look. I hope a rebase on the latest Tuya core integration code will be enough, but as of now I cannot guarantee anything. For the time being, please revert to 2025.02 if you need the integration to work.

**Comment by Elhoj** [comment]
Hi @AlessandroTischer - are you also awair of the below future issue?
(And by the way many thanks for all you help/work)
------------------------------------
Logger: homeassistant.components.climate
Source: components/climate/__init__.py:618
integration: Climate (documentation, issues)
First occurred: March 8, 2025 at 20:33:07 (8 occurrences)
Last logged: 12:44:43

tuya_climate_mult_edit::TuyaClimateEntity sets the hvac_mode cool which is not valid for this entity with modes: off, heat_cool, heat_cool. This will **stop working in 2025.4** and raise an error instead. Please report it to the author of the 'tuya_climate_mult_edit' custom integration
------------------------------------

**And** (Not that I am using a Vacuum or Light integration)

------------------------------------
Logger: homeassistant.components.vacuum
Source: helpers/deprecation.py:222
integration: Vacuum (documentation, issues)
First occurred: March 8, 2025 at 19:26:15 (6 occurrences)
Last logged: March 8, 2025 at 19:26:15

STATE_CLEANING was used from tuya_climate_mult_edit, this is a deprecated constant which will **be removed in HA Core 2026.1**. Use VacuumActivity.CLEANING instead, please report it to the author of the 'tuya_climate_mult_edit' custom integration
STATE_DOCKED was used from tuya_climate_mult_edit, this is a deprecated constant which will be removed in HA Core 2026.1. Use VacuumActivity.DOCKED instead, please report it to the author of the 'tuya_climate_mult_edit' custom integration
STATE_RETURNING was used from tuya_climate_mult_edit, this is a deprecated constant which will be removed in HA Core 2026.1. Use VacuumActivity.RETURNING instead, please report it to the author of the 'tuya_climate_mult_edit' custom integration
-----------------------------------
Logger: homeassistant.components.light
Source: helpers/deprecation.py:222
integration: Light (documentation, issues)
First occurred: March 8, 2025 at 19:26:15 (2 occurrences)
Last logged: March 8, 2025 at 19:26:15

ATTR_COLOR_TEMP was used from tuya_climate_mult_edit, this is a deprecated constant which will **be removed in HA Core 2026.1**. Use kelvin equivalent (ATTR_COLOR_TEMP_KELVIN) instead, please report it to the author of the 'tuya_climate_mult_edit' custom integration
------------------------------------

Best regards

**Comment by AlessandroTischer** [comment]
Hello everyone! I just updated the sources using the latest core component. It should work now, when using HA 2025.03. 

Be careful, I'm not sure the new version works with older HA versions!

**Comment by amoranvalero** [comment]
> Hello everyone! I just updated the sources using the latest core component. It should work now, when using HA 2025.03. 
> 
> Be careful, I'm not sure the new version works with older HA versions!

Working back again! Thanks a lot

**Comment by amoranvalero** [comment]
Hello @AlessandroTischer can you reopen the issue with the last update 2025.5.2 I am experimenting the same issue. Your Integration do not repot back the status of the entities.can you along to tuya core HA integration. Sorry for messing with the regular demands of alignment but there is definitely something buggy when core integration get updated.
